features/nanostack/mbed-mesh-api/source/wisun_tasklet.c

```diff
@@ -336,5 +364,54 @@ int8_t wisun_tasklet_network_init(int8_t device_id)
     if (!mac_api) {
         mac_api = ns_sw_mac_create(device_id, &storage_sizes);
     }
+
+    if (!wisun_settings_str.network_name) {
+        // No network name set by API, use network name from configuration
+        int wisun_network_name_len = sizeof(MBED_CONF_MBED_MESH_API_WISUN_NETWORK_NAME);
+        wisun_settings_str.network_name = (char *)ns_dyn_mem_alloc(wisun_network_name_len);
+        if (!wisun_settings_str.network_name) {
+            return -3;
+        }
+        strncpy(wisun_settings_str.network_name, MBED_CONF_MBED_MESH_API_WISUN_NETWORK_NAME, wisun_network_name_len);
+    }
+
     return arm_nwk_interface_lowpan_init(mac_api, INTERFACE_NAME);
 }
+
+int wisun_tasklet_network_name_set(int8_t nwk_interface_id, char *network_name_ptr)
+{
+    if (!network_name_ptr || strlen(network_name_ptr) > 32) {
+        return -1;
+    }
+
+    // save the network name to have support for disconnect/connect
+    ns_dyn_mem_free(wisun_settings_str.network_name);
+    wisun_settings_str.network_name = (char *)ns_dyn_mem_alloc(strlen(network_name_ptr) + 1);
+    if (!wisun_settings_str.network_name) {
+        return -2;
+    }
+
+    strcpy(wisun_settings_str.network_name, network_name_ptr);
+
+    if (wisun_tasklet_data_ptr && wisun_tasklet_data_ptr->tasklet_state == TASKLET_STATE_BOOTSTRAP_READY) {
```

Comment:
this can be called any time after interface up even during bootstrap. I think in here you should have flexibility for it even before bootstrap is ready